server/rest.go

```diff
@@ -1210,6 +1221,138 @@ func (s *Server) ContractEvents(c echo.Context) error {
+		if errImage != nil {
+			return nil, errImage
+		}
+		break
+	case strings.Contains(rawString, "data:image/jpeg;base64,"):
+		rawString = strings.ReplaceAll(rawString, "data:image/jpeg;base64,", "")
+		unbased, _ = base64.StdEncoding.DecodeString(string(rawString))
+		imageDecode, errImage = jpeg.Decode(bytes.NewReader(unbased))
+		if errImage != nil {
+			return nil, errImage
+		}
+		break
+	case strings.Contains(rawString, "data:image/webp;base64"):
+		rawString = strings.ReplaceAll(rawString, "data:image/webp;base64,", "")
+		unbased, _ = base64.StdEncoding.DecodeString(string(rawString))
+		imageDecode, errImage = webp.Decode(bytes.NewReader(unbased))
+		if errImage != nil {
+			return nil, errImage
+		}
+		break
+	default:
+		break
+	}
+
+	return imageDecode, nil
+}
+
+func EncodeImage(image image.Image, rawString string, fileName string) ([]byte, string) {
+	switch {
+	case strings.Contains(rawString, "data:image/png;base64,"):
+		buf := new(bytes.Buffer)
+		errConverter := png.Encode(buf, image)
+		if errConverter != nil {
+			return nil, ""
+		}
+		sendS3 := buf.Bytes()
+		spl := strings.Split(fileName+".png", ".")
+		uploadedFileName := strings.Join(spl, ".")
+		return sendS3, uploadedFileName
+	case strings.Contains(rawString, "data:image/jpeg;base64,"):
+		buf := new(bytes.Buffer)
+		errConverter := jpeg.Encode(buf, image, nil)
+		if errConverter != nil {
+			return nil, ""
+		}
+		sendS3 := buf.Bytes()
+		spl := strings.Split(fileName+".png", ".")
+		uploadedFileName := strings.Join(spl, ".")
+		return sendS3, uploadedFileName
+	case strings.Contains(rawString, "data:image/webp;base64"):
+		buf := new(bytes.Buffer)
+		errConverter := webp.Encode(buf, image, nil)
+		if errConverter != nil {
+			return nil, ""
+		}
+		sendS3 := buf.Bytes()
+		spl := strings.Split(fileName+".webp", ".")
+		uploadedFileName := strings.Join(spl, ".")
+		return sendS3, uploadedFileName
+	}
+
+	return nil, ""
+}
+
+func UploadLogo(c echo.Context, rawString string, fileName string, session *session.Session) (string, error) {
+	uploader := s3manager.NewUploader(session)
+
+	if strings.Contains(rawString, "https") && (strings.Contains(rawString, "png") || strings.Contains(rawString, "jpeg") || strings.Contains(rawString, "webp")) {
+		return rawString, nil
+	}
+
+	fileUpload, err := Base64ToImage(rawString)
+	if err != nil {
+		return "", api.Invalid.Build(c)
+	}
+
+	sendS3, uploadedFileName := EncodeImage(fileUpload, rawString, fileName)
+
+	_, newErr := uploader.Upload(&s3manager.UploadInput{
+		Bucket: aws.String("cdn1.bcms.tech"),
+		ACL:    aws.String("public-read"),
+		Key:    aws.String("/kai-explorer-backend/logo/" + uploadedFileName),
+		Body:   bytes.NewReader(sendS3),
+	})
+
+	if newErr != nil {
+		return "", api.Invalid.Build(c)
+	}
+	pathAvatar := "https://s3-ap-southeast-1.amazonaws.com/cdn1.bcms.tech/kai-explorer-backend/logo/"
+	filepath := pathAvatar + uploadedFileName
+
+	return filepath, nil
+}
```

Comment:
why we still keep c echo.Context as args ?

Reply:
I removed it